eva-accession-import-automation/genome_downloader.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2020 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from get_assembly_report_url import *
+import os
+import pandas
+import sys
+import urllib.request
+import re
+import json
+from __init__ import *
+
+
+def get_header_line_index(assembly_report_path):
+    try:
+        with open(assembly_report_path) as assembly_report_file_handle:
+            return next(line_index for line_index, line in enumerate(assembly_report_file_handle)
+                        if line.lower().startswith("# sequence-name") and "sequence-role" in line.lower())
+    except StopIteration:
+        raise Exception("Could not locate header row in the assembly report!")
+
+
+def get_dataframe_for_assembly_report(assembly_report_path):
+    return pandas.read_csv(assembly_report_path, skiprows=get_header_line_index(assembly_report_path),
+                           dtype=str, sep='\t')
+
+
+def download_assembly_report(assembly_accession, directory_path):
+    assembly_report_url = get_assembly_report_url(assembly_accession)
+    assembly_report_path = directory_path + os.path.basename(assembly_report_url)
+    os.makedirs(os.path.dirname(assembly_report_path), exist_ok=True)
+    urllib.request.urlretrieve(assembly_report_url, assembly_report_path)
+    urllib.request.urlcleanup()
+    return assembly_report_path
+
+
+def get_sequence_from_ncbi(accession, fasta_path, eutils_api_key, written_contigs):
+    url = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nuccore&id=' + accession + \
+           '&rettype=fasta&retmode=text&api_key=' + eutils_api_key + '&tool=eva&email=eva-dev@ebi.ac.uk'
+    logger.info('Downloading ' + accession)
+    with urllib.request.urlopen(url) as response:
```

Comment:
Have you checked memory usage for spectacularly large contigs? The bash script used for dbSNP import side-stepped this issue by [redirecting wget output to a file](https://github.com/EBIvariation/eva-accession/blob/b549929622d207dfdfae7c0967e934b0221bc5da/eva-accession-import-automation/create_fasta_from_assembly_report.sh#L54)

Reply:
What happens for requests that fail? IMO, response code should be inspected first...